prototypes/ScheduledMerges.hs

```diff
@@ -781,8 +795,11 @@ update tr (LSMHandle scr lsmr) k op = do
     let wb' = Map.insertWith combine k op wb
     if bufferSize wb' >= maxBufferSize
       then do
-        ls' <- increment tr sc (bufferToRun wb') ls unionLevel
-        let content' = LSMContent Map.empty ls' unionLevel
+        -- Before adding the run to the regular levels, we check if we can get
+        -- rid of the union level (by moving it into into the regular ones).
```

Comment:
Okay, so this is the approach where we try to put the completed union run into the levels whenever we flush a write buffer. What were the other alternatives? Could this not be implemented as part of `supplyUnionCredits`?

Reply:
Ah, yes, now I remember. So one such situation would be where we create an incremental union table, duplicate it, and then only supply credits to one of them. Would it work if the other duplicate (which did not supply credits) always has 1 credit remaining for the "moving back into the levels'?

Reply:
Yes, could be due to a duplicate or also using the table as an input to another union.

> Would it work if the other duplicate (which did not supply credits) always has 1 credit remaining for the "moving back into the levels'?

That would help make it clear that `supplyUnionCredits` still *should* be called, but it still doesn't guarantee it.

Reply:
It feels like if we rely on `updates` for the union level to be migrated, that it is also not fully guaranteed that it will migrate, or is it?